### PR TITLE
Circle CI error handling in test.js

### PR DESCRIPTION
**Pull request by chrisberthe**
Revert changes made in #72 and instead exit the process - learned that `throw new Error()` is not enough to exit Circle CI from running.

### CODE REVIEW

File: src/tasks/lint-locales.js [hunk 1]
```diff
@@ -12,9 +12,16 @@ const Reporter = require('./includes/lint-reporter.js').default;
...
 function lintLocales() {
-  return runAll(config.src.root, new Reporter()).then((reporter) => {
-    return reporter.output();
-  });
+  return runAll(config.src.root, new Reporter())
+    .then((reporter) => {
+      return reporter.output();
+    })
+    .catch((err) => {
+      console.log(err);
```
Comment (NathanPJF) [inline src/tasks/lint-locales.js]: This is the only `console.*` in code base.  Mind switching to `gutil` for consistency?

**Comment by davidcornu** [comment]
Looks like we call `spawn` [in a bunch of other places](https://github.com/Shopify/slate-tools/search?utf8=%E2%9C%93&q=spawn&type=) without registering a `close` handler, which means tasks like `build` or `deploy` could fail with a `0` exit code. 

In a follow-up PR, we should make sure the behaviour is consistent everywhere.

**Comment by davidcornu** [comment]
I noticed that we're committing a `package-lock.json`. According to the docs (https://docs.npmjs.com/files/package-lock.json), these can't be published to NPM. We could however, include a `npm-shrinkwrap.json` (https://docs.npmjs.com/files/shrinkwrap.json) but

> The recommended use-case for npm-shrinkwrap.json is applications deployed through the publishing process on the registry: for example, daemons and command-line tools intended as global installs or devDependencies. It's strongly discouraged for library authors to publish this file, since that would prevent end users from having control over transitive dependency updates.

To avoid confusion I'd recommend we add both `package-lock.json` and `npm-shrinkwrap.json` to `.gitignore`, and remove the `package-lock.json` from this PR.

**Comment by chrisberthe** [comment]
Thanks for noticing the `package-lock.json` @davidcornu - didn't have any intention of adding that to this PR.

File: src/tasks/lint-locales.js (hunk 1 above)
Comment (chrisberthe) [inline src/tasks/lint-locales.js]: Yeah I felt weird about that too / was going to ask you your opinion - thanks for pointing that out.

File: src/commands/test.js [hunk 2]
```diff
@@ -15,6 +15,13 @@ export default function(program) {
       spawn(config.gulp, ['test', '--gulpfile', config.gulpFile, '--cwd', config.themeRoot], {
         detached: false,
         stdio: 'inherit',
+      }).on('close', (code) => {
+        if (code === 0) {
+          return;
+        }
+
+        const exitCode = 2;
```
Comment (t-kelly) [inline src/commands/test.js]: Do we need this `const`? Why not just `process.exit(2);`

File: src/commands/test.js [hunk 3]
```diff
@@ -15,6 +15,13 @@ export default function(program) {
       spawn(config.gulp, ['test', '--gulpfile', config.gulpFile, '--cwd', config.themeRoot], {
         detached: false,
         stdio: 'inherit',
+      }).on('close', (code) => {
+        if (code === 0) {
```
Comment (t-kelly) [inline src/commands/test.js]: All one line: `if (code === 0) { return;}`

File: src/tasks/lint-locales.js [hunk 4]
```diff
@@ -12,9 +13,16 @@ const Reporter = require('./includes/lint-reporter.js').default;
  * @private
  */
 function lintLocales() {
-  return runAll(config.src.root, new Reporter()).then((reporter) => {
-    return reporter.output();
-  });
+  return runAll(config.src.root, new Reporter())
+    .then((reporter) => {
```
Comment (t-kelly) [inline src/tasks/lint-locales.js]: Could alternatively be written: `.then(reporter => reporter.output())`

File: src/tasks/lint-locales.js [hunk 5]
```diff
@@ -12,9 +13,16 @@ const Reporter = require('./includes/lint-reporter.js').default;
...
+      return reporter.output();
+    })
+    .catch((err) => {
+      gutil.log(err);
+
+      const exitCode = 2;
```
Comment (t-kelly) [inline src/tasks/lint-locales.js]: Same here as mentioned above.

File: src/commands/test.js (hunk 2 above)
Comment (chrisberthe) [inline src/commands/test.js]: We don't - it was just a pattern that was already there in other parts of the codebase - I'll change this for all instances.

**Pull request merged**